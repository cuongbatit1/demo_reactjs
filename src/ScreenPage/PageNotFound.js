import React from "react";
import PropTypes from "prop-types";
// import imageRes from "./../../assets/Black_Hole_Orange.png";
const imageRes = "/static/images/Black_Hole_Orange.png";

const PageNotFound = (props) => {
  const style404 = {
    fontSize: 72,
    fontWeight: "bold",
  };

  let message = "Page not found";
  if (props.text) message = props.text;

  return (
    <div style={{ textAlign: "center" }}>
      <h1>{message}</h1>
      <div style={style404}>404</div>
      <img src={imageRes} alt="" />
    </div>
  );
};

PageNotFound.propTypes = {
  text: PropTypes.string,
};

export default PageNotFound;
