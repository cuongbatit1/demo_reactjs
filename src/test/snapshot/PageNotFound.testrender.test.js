import { render, screen } from "@testing-library/react";
import PageNotFound from "../../ScreenPage/PageNotFound";

test("test renders PageNotFound snapshot", () => {
  const { asFragment } = render(<PageNotFound />);
  expect(asFragment()).toMatchSnapshot();
});
