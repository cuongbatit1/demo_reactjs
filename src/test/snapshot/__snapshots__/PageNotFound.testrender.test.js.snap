// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders PageNotFound snapshot 1`] = `
<DocumentFragment>
  <div
    style="text-align: center;"
  >
    <h1>
      Page not found
    </h1>
    <div
      style="font-size: 72px; font-weight: bold;"
    >
      404
    </div>
    <img
      alt=""
      src="/static/images/Black_Hole_Orange.png"
    />
  </div>
</DocumentFragment>
`;
